src/lib/firebase.js

```diff
@@ -0,0 +1,92 @@
+import { initializeApp } from "firebase/app";
+import { initializeAppCheck, ReCaptchaV3Provider } from "firebase/app-check";
+import { getToken } from "firebase/app-check";
+import { UAParser } from "ua-parser-js";
+
+const firebaseConfig = {
+  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
+  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
+  appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
+};
+
+export const app = initializeApp(firebaseConfig);
+
+let appCheckInstance = null;
+
+const getAppCheck = () => {
+  if (!appCheckInstance) {
+    try {
+      appCheckInstance = initializeAppCheck(app, {
+        provider: new ReCaptchaV3Provider(process.env.NEXT_PUBLIC_RECAPTCHA_KEY),
+        isTokenAutoRefreshEnabled: true,
+      });
+    } catch (error) {
+      console.error("Error initializing app check", error);
+    }
+  }
+  
+  return appCheckInstance;
+};
+
+export async function sendSurveyFeedback(answers) {
+  try {
+    const appCheck = getAppCheck();
+    
+    if (!appCheck) {
+      throw new Error("App Check not initialized");
+    }
+
+    const tokenResult = await getToken(appCheck);
+
+    const parser = new UAParser().getResult();
+
+    const browserInfo = {
+      name: parser.browser.name,
+      version: parser.browser.version
+    };
+
+    const screenInfo = {
+      width: window.screen.width,
+      height: window.screen.height
+    };
+
+    const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+    const resp = await fetch("https://api.ipify.org?format=json");
+    const data = await resp.json();
+    const userId =  data.ip; 
```

Comment:
Since the IP isn't a data that we want to have the responsibility over, it's better if we remove it.
```suggestion
```